basyx/aas/model/submodel.py

```diff
@@ -1176,6 +1171,9 @@ def _validate_asset_ids_for_entity_type(entity_type: base.EntityType,
             raise base.AASConstraintViolation(
                 14, "A co-managed entity has to have neither a globalAssetId nor a specificAssetId")
 
+        if global_asset_id is not None:
+            _string_constraints.check_identifier(global_asset_id)
```

Comment:
I don't like that this is validated every time only the `specific_asset_id` changes, IMO it would be prettier to just verify this in the `global_asset_id` setter and `__init__`. Same for `AssetInformation`.